src/App.tsx

```diff
@@ -1,26 +1,89 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, deleteTodos, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
+import { TodoStatusFilter } from './types/TodoStatusFilter';
+import { getPreparedTodos } from './utils/getPreparedTodos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState([0]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState<TodoStatusFilter>(
+    TodoStatusFilter.All,
+  );
+
+  const activeTodosCount = todos.filter(todo => !todo.completed).length;
+  const completedTodosCount = todos.filter(todo => todo.completed).length;
+  const preparedTodos = getPreparedTodos(todos, selectedFilter);
+
+  const handleDeleteTodo = (todoId: number) => {
+    setLoading(prev => [...prev, todoId]);
+
+    deleteTodos(todoId)
+      .then(() =>
+        setTodos((prev: Todo[]) => prev.filter(todo => todo.id !== todoId)),
+      )
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+      })
+      .finally(() => {
+        setLoading((prev: number[]) => prev.filter(id => id !== todoId));
+      });
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
```

Comment:
let's create enum with error messages and use its in every place where toy use magic string)